-) Implemented local database (sqlite) running option

itsdeka opened the pull request:
-) Updated requirements.txt
-) Changed EventEmitter(asyncio) with AsyncIOEventEmitter
-) Implemented exe.with_local_database()

### Description:
Implemented a new way to fetch (and store locally) candles from REST API

### Breaking changes:
- [ ]

### New features:
- [X]

### Fixes:
- [X]

### PR status:
- [X] Version bumped
- [X] Change-log updated

itsdeka committed:
-) Implemented local database (sqlite) running option
-) Updated requirements.txt
-) Changed EventEmitter(asyncio) with AsyncIOEventEmitter

itsdeka committed:
-) Updated README.md

itsdeka committed:
-) Updated README.md

itsdeka committed:
fixed bug in retrieving candles

itsdeka committed:
fixed bug in retrieving candles

itsdeka committed:
updated setup.py and CHANGELOG

itsdeka commented:
@robertkowalski

robertkowalski commented on examples/local_database.py:
```diff
@@ -0,0 +1,57 @@
+then = now - (1000 * 60 * 60 * 24 * 36) # 5 days ago
+
+loop = asyncio.get_event_loop()
+loop.run_until_complete(exe.with_local_database(then, now))
```
missing newline on end of file

robertkowalski commented on hfstrategy/utils/db.py:
```diff
@@ -0,0 +1,168 @@
+from bfxapi import Client
+
+db = SqliteDatabase('bfx-hf-strategy.db')
+db.connect()
```
hm, can that get moved to the constructor, so its just called and set up when someone uses the DB class?

itsdeka replied:
I need to assign db to the database variable within the class Meta of BaseModel()

Do you think is it ok if we keep only the first line here? 
It does not generate any file when called. And I move db.create_tables() in a specific function initialize_db()

robertkowalski commented on hfstrategy/utils/db.py:
```diff
@@ -0,0 +1,168 @@
+    def to_list(self):
+        return [self.mts, self.open, self.close, self.high, self.low, self.volume]
+
+db.create_tables([Candle])
```
same here...

itsdeka replied:
ty

robertkowalski commented on requirements.txt:
```diff
@@ -10,3 +10,6 @@ numpy==1.15.4
 # websocket-client==0.54.0
+bitfinex-api-py==1.1.10
+peewee==3.14.2
+nest-asyncio==1.5.1
```
> pip3 install -r requirements.txt

```
ERROR: Cannot install bitfinex-api-py==1.1.10 and pyee==5.0.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested pyee==5.0.0
    bitfinex-api-py 1.1.10 depends on pyee~=8.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

itsdeka replied:
version of pyee updated to 8.0.1 in requirements

robertkowalski reviewed (changes requested):
code works very nice, thanks!

found 3 small issues that need to get addressed

itsdeka committed:
-) Final empty line added
-) (FIX) the sqlite database is now generated only when needed, db = SqliteDatabase('') initializes the class but does not produce the db

itsdeka committed:
fixed pyee version in requirements.txt

robertkowalski reviewed (approved):
lgtm
